packages/astro/src/assets/build/generate.ts

```diff
@@ -269,6 +294,17 @@ export async function generateImagesForPath(
 	}
 }
 
+async function writeRemoteCacheFile(cachedFileURL: URL, resultData: ImageData) {
+	return await fs.promises.writeFile(
+		cachedFileURL,
+		JSON.stringify({
+			data: Buffer.from(resultData.data).toString('base64'),
+			expires: resultData.expires,
+			etag: resultData.etag,
+		}),
```

Comment:
I've wrapped this in a try catch block which will print a warning and no-op, rather than throwing an Error. The asset should just skip the cache in this case.